core/tests/protractor-browserstack.conf.js

```diff
@@ -0,0 +1,219 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Protractor configuration to run tests on Browserstack.
+ */
+
+var browserstack = require('browserstack-local');
+var dotenv = require('dotenv');
+
+// eslint-disable-next-line angular/di
+var result = dotenv.config({path: 'core/tests/.browserstack.env'});
+
+// A reference configuration file.
+exports.config = {
+  // ----- How to setup Selenium -----
+  //
+  // There are three ways to specify how to use Selenium. Specify one of the
+  // following:
+  //
+  // 1. seleniumServerJar - to start Selenium Standalone locally.
+  // 2. seleniumAddress - to connect to a Selenium server which is already
+  //    running.
+  // 3. sauceUser/sauceKey - to use remote Selenium servers via SauceLabs.
+  //
+  // If the chromeOnly option is specified, no Selenium server will be started,
+  // and chromeDriver will be used directly (from the location specified in
+  // chromeDriver)
+
+  // The location of the selenium standalone server .jar file, relative
+  // to the location of this config. If no other method of starting selenium
+  // is found, this will default to
+  // node_modules/protractor/selenium/selenium-server...
+  seleniumServerJar: null,
+  // The port to start the selenium server on, or null if the server should
+  // find its own unused port.
+  seleniumPort: null,
+  // Chromedriver location is used to help the selenium standalone server
+  // find chromedriver. This will be passed to the selenium jar as
+  // the system property webdriver.chrome.driver. If null, selenium will
+  // attempt to find chromedriver using PATH.
+  chromeDriver: './selenium/chromedriver',
+  // If true, only chromedriver will be started, not a standalone selenium.
+  // Tests for browsers other than chrome will not run.
+  chromeOnly: false,
+  // Additional command line options to pass to selenium. For example,
+  // if you need to change the browser timeout, use
+  // seleniumArgs: ['-browserTimeout=60'],
+  seleniumArgs: [],
+
+  // If sauceUser and sauceKey are specified, seleniumServerJar will be ignored.
+  // The tests will be run remotely using SauceLabs.
+  sauceUser: null,
+  sauceKey: null,
+
+  // The address of a running selenium server. If specified, Protractor will
+  // connect to an already running instance of selenium. This usually looks like
+  // seleniumAddress: 'http://localhost:4444/wd/hub'
+  seleniumAddress: 'http://hub-cloud.browserstack.com/wd/hub',
+
```

Comment:
I can see this is from https://www.browserstack.com/automate/node 
Should we complete remove the `seleniumAddress: 'http://localhost:4444/wd/hub'` line then since it is not needed?

Reply:
The above line says: `This usually looks..`. I don't think that comment is wrong since we still have an address ending with `/wd/hub`.